### ISSUE TITLE
Indentation: method definition not checked when preceded by annotation

### ISSUE THREAD

**Issue by rnveach** [checkstyle/checkstyle#3130]
````
$ cat TestClass.java
public class TestClass {
    @SuppressWarnings( //indent:4 exp:4
        value="" //indent:8 exp:8
    ) //indent:4 exp:4
public void testStartOfSequence1() { //indent:0
    } //indent:4 exp:4

    @SuppressWarnings(value="") //indent:4 exp:4
public void testStartOfSequence2() { //indent:0
    } //indent:4 exp:4

public void testStartOfSequence3() { //indent:0
    } //indent:4 exp:4
}

$ cat TestConfig.xml
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
          "-//Puppy Crawl//DTD Check Configuration 1.3//EN"
          "http://www.puppycrawl.com/dtds/configuration_1_3.dtd">

<module name="Checker">
    <property name="charset" value="UTF-8"/>

    <module name="TreeWalker">
<property name="tabWidth" value="4"/>

    <module name="Indentation">
      <property name="basicOffset" value="4"/>
      <property name="braceAdjustment" value="0"/>
      <property name="lineWrappingIndentation" value="4"/>
    </module>
    </module>
</module>

$ java -jar checkstyle-6.17-all.jar -c TestConfig.xml TestClass.java
Starting audit...
[ERROR] TestClass.java:12: 'method def modifier' have incorrect indentation level 0, expected level should be 4. [Indentation]
Audit done.
Checkstyle ends with 1 errors.
````

All 3 method definitions start at column 0, but only the one with no annotation before it is flagged as a violation.
I was expecting lines 5 and 9 to be flagged too as method definitions should start at column 4, not 0, in this scenario.

**Comment by rnveach** [comment]
It seems the issue is we only look at the modifier group for where the method definition should be placed. The problem with annotations is that they are considered modifiers and hence are added to the modifier group. The group then takes on the line number of the annotations so all we end up checking is the annotation line, not the method definition line. If the annotation was incorrectly placed, we would display the modifier error message `'method def modifier' have incorrect indentation`.

Example:
Code: https://github.com/checkstyle/checkstyle/blob/ccdc3c09c500ccd31d56e4d6ed9c2d86edf2e9f1/src/test/resources/com/puppycrawl/tools/checkstyle/grammars/InputRegressionJavaClass1.java#L79-L80
Ast: https://github.com/checkstyle/checkstyle/blob/ccdc3c09c500ccd31d56e4d6ed9c2d86edf2e9f1/src/test/resources/com/puppycrawl/tools/checkstyle/grammars/InputRegressionJavaClass1Ast.txt#L608-L616
Modifiers group says line 79 while `public` is on line 80.

**Comment by romani** [comment]
Almost completed but abandoned update https://github.com/checkstyle/checkstyle/pull/9195#issuecomment-1836963836 , everyone welcome to reuse this code to finish a fix
